src/server.c

```diff
@@ -2704,7 +2704,8 @@ void commandAddSubcommand(struct redisCommand *parent, struct redisCommand *subc
 void setImplicitACLCategories(struct redisCommand *c) {
     if (c->flags & CMD_WRITE)
         c->acl_categories |= ACL_CATEGORY_WRITE;
-    if (c->flags & CMD_READONLY)
+    /* RO script commands don't have the read category */
+    if (c->flags & CMD_READONLY && c->proc != evalRoCommand && c->proc != fcallroCommand)
```

Comment:
missing `evalShaRoCommand`.
maybe check the `ACL_CATEGORY_SCRIPTING` instead?

Reply:
I ended up going with this (sorry I got distracted part way through updating the PR), I also fixed up the tests.